### PR TITLE
(db/snapshots): Integration and Enhancement

### PR DESCRIPTION
**Pull request by XuPeng-SH**
**What type of PR is this?**

api-change / bug / improvement / documentation / feature

**Need this PR be picked to master branch?**

Yes / No

**Which issue(s) this PR fixes:**

Fixes #

**What this PR does / why we need it:**

**Special notes for your reviewer:**

**Additional documentation (e.g. design docs, usage docs, etc.):**

### CODE REVIEW

File: core/src/db/DB.h [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
 #pragma once
 
+#include <map>
```
Comment (cydrain) [inline core/src/db/DB.h:14]: We'd better create new files for DB.h / DBImpl.h / DBImpl.cpp

File: core/src/db/DB.h [hunk 2]
```diff
@@ -48,21 +52,32 @@ class DB {
 
     virtual Status
     CreateCollection(meta::CollectionSchema& table_schema_) = 0;
+    virtual Status
+    SSTODOCreateCollection(const snapshot::CreateCollectionContext& context) = 0;
```
Comment (cydrain) [inline core/src/db/DB.h:56]: what SSTODO means ?

File: core/src/db/DB.h (hunk 1 above)
Comment (XuPeng-SH) [reply core/src/db/DB.h:14]: This PR is a placeholder for CI health check. The integration approach is open and will be discussed after the holiday.

File: core/src/db/DBImpl.cpp [hunk 3]
```diff
@@ -3329,5 +3332,185 @@ DBImpl::ResumeIfLast() {
...
+
+Status
+DBImpl::SSTODOHasCollection(const std::string& collection_name, bool& has_or_not) {
+    if (!initialized_.load(std::memory_order_acquire)) {
+        return SHUTDOWN_ERROR;
+    }
+
+    snapshot::ScopedSnapshotT ss;
+    auto status = snapshot::Snapshots::GetInstance().GetSnapshot(ss, collection_name);
+    if (!status.ok()) {
```
Comment (cydrain) [inline core/src/db/DBImpl.cpp:3402]: has_or_not = status.ok();

File: core/src/db/DB.h (hunk 1 above)
Comment (cydrain) [inline core/src/db/DB.h:14]: removable

File: core/src/db/SSDBImpl.cpp [hunk 4]
```diff
@@ -49,5 +72,169 @@ SSDBImpl::Stop() {
...
+        fields_schema[kv.second.Get()] = ss->GetFieldElementsByField(kv.second->GetName());
+    }
+    return status;
+}
+
+Status
+SSDBImpl::DropCollection(const std::string& name) {
+    CHECK_INITIALIZED;
+
+    // dates partly delete files of the collection but currently we don't support
```
Comment (cydrain) [inline core/src/db/SSDBImpl.cpp:115]: this comment seems not readable

File: core/src/db/SnapshotHandlers.cpp [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    : BaseT(ss), context_(context) {
+}
+
+Status
+LoadVectorFieldHandler::Handle(const snapshot::FieldPtr& field) {
+    if (field->GetFtype() != snapshot::FieldType::VECTOR) {
+        return Status::OK();
+    }
+    if (context_ && context_->IsConnectionBroken()) {
+        LOG_ENGINE_DEBUG_ << "Client connection broken, stop load collection";
```
Comment (cydrain) [inline core/src/db/SnapshotHandlers.cpp:47]: stop loading

File: core/src/db/SnapshotHandlers.cpp [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+        return Status(DB_ERROR, "Connection broken");
+    }
+
+    // SS TODO
+    auto element_handler = std::make_shared<LoadVectorFieldElementHandler>(context_, ss_, field);
+    element_handler->Iterate();
```
Comment (cydrain) [inline core/src/db/SnapshotHandlers.cpp:53]: it's not clear what "Iterate()" does internally.
Is it possible write the code in this way:
auto elem_iter = element_handler.begin();
while (elem_iter) {
    /* do something */
    elem_iter = element_handler.next();
}

File: core/src/db/snapshot/ResourceTypes.h [hunk 7]
```diff
@@ -24,8 +24,10 @@ using NUM_TYPE = int64_t;
 using FTYPE_TYPE = int64_t;
 using TS_TYPE = int64_t;
 using LSN_TYPE = uint64_t;
```
Comment (cydrain) [inline core/src/db/snapshot/ResourceTypes.h:26]: suggest to use int64_t everywhere for ease of debug

**Review by cydrain — APPROVED** [review]

**Review by wxyucs — APPROVED** [review]

**Pull request merged**